frontend/src/lib/getChartData.ts

```diff
@@ -6,6 +6,16 @@ import convertIntoEpoc from './covertIntoEpoc';
 import { colors } from './getRandomColor';
 
 const getChartData = ({ queryData }: GetChartDataProps): ChartData => {
+	const uniqueTimeLabels = new Set<number>();
+	queryData.forEach((data) => {
+		data.queryData.forEach((query) => {
+			query.values.forEach((value) => {
+				uniqueTimeLabels.add(value[0]);
```

Comment:
That list contains only a partial list of values so we have it outside to make a full list of timestamps of combined items.

Reply:
I have another question about this though how this behaves with multiple queries in the same dashboard?

Reply:
In that case, this should be fine. For a query, there can be missing and uneven points which were creating the reported bug. This fixes it by creating the full list of those timestamps and filling the mission values with null so the line chart works fine with setting `spanGaps` enabled.

Reply:
>we are already doing the n2 below maybe we can hop in [here](https://github.com/SigNoz/signoz/blob/develop/frontend/src/lib/getChartData.ts#L12)

No, we need the list beforehand because it's used right below to fill the missing values in `filledDataValues` so moving it there will still result in partial values.

Reply:
the graph data changes quite frequently so I think it's not a good idea to memorize this.